src/shogun/util/factory.h

```diff
@@ -335,14 +338,9 @@ namespace shogun
 		return std::make_shared<PipelineBuilder>();
 	}
 	} // namespace details
-	// fix me
-	//#ifdef SWIG
+
 	template <typename TypeName, typename... Args>
 	std::shared_ptr<TypeName> create(Args... args)
```

Comment:
Can you put the perfect forwarding now and hide this whole function from swig? I’m thinking that swig doesn’t need to see any of this as the code that relies on this is written as literal cpp code (using %{ %})

Reply:
but there are many `creat_` function in swig need the `create<>`, such as `%template(create_loss) shogun::create<shogun::LossFunction, std::string>;
`

Reply:
maybe we should write a wrapper function named _create
```cpp
template <typename TypeName, typename... Args>
std::shared_ptr<TypeName> _create(Args... args)
{ return create<TypeName>(args...);}
```
then use `_create` in swig and use `create` in c++ side.

Reply:
ah yes of course, I thought everything was pasted in swig generated code with `%{ %}`. Yes, the wrapper function would be good if it works. `create` for C++ and `create_` for SWIG, and hide `create_` in detail namespace or something.. :)